client/payment-methods-map.tsx

```diff
@@ -29,17 +31,39 @@ import {
 
 const accountCountry = window.wcpaySettings?.accountStatus?.country || 'US';
 
-export interface PaymentMethodMapEntry {
-	id: string;
-	label: string;
-	description: string;
-	icon: ReactImgFuncComponent;
-	currencies: string[];
-	stripe_key: string;
-	allows_manual_capture: boolean;
-	allows_pay_later: boolean;
-	accepts_only_domestic_payment: boolean;
-}
+import type { PaymentMethodMapEntry } from './types/payment-methods';
+
+// Get any payment method definitions from the client.
+const PaymentMethodDefinitions =
+	typeof woopaymentsPaymentMethodDefinitions !== 'undefined'
+		? woopaymentsPaymentMethodDefinitions
+		: {};
+
+const convertedPaymentMethodDefinitions = Object.fromEntries(
+	Object.entries( PaymentMethodDefinitions ).map( ( [ key, value ] ) => [
+		key,
+		{
+			id: value.id,
+			label: value.title,
+			description: value.description,
+			icon: ( { className } ) => (
+				<img
+					src={ value.icon }
+					alt={ value.title }
+					className={ classNames(
+						'payment-method__icon',
+						className
+					) }
+				/>
+			),
+			currencies: value.currencies,
+			stripe_key: value.stripeId,
+			allows_manual_capture: value.allows_manual_capture,
+			allows_pay_later: value.allows_pay_later,
+			accepts_only_domestic_payment: value.accepts_only_domestic_payment,
+		} as PaymentMethodMapEntry,
+	] )
+);
```

Comment:
You could do something like this, instead of casting with `as`
```suggestion
const convertedPaymentMethodDefinitions = Object.fromEntries<
	PaymentMethodMapEntry
>(
	Object.entries( PaymentMethodDefinitions ).map( ( [ key, value ] ) => [
		key,
		{
			id: value.id,
			label: value.title,
			description: value.description,
			icon: ( { className } ) => (
				<img
					src={ value.icon }
					alt={ value.title }
					className={ classNames(
						'payment-method__icon',
						className
					) }
				/>
			),
			currencies: value.currencies,
			stripe_key: value.stripeId,
			allows_manual_capture: value.allows_manual_capture,
			allows_pay_later: value.allows_pay_later,
			accepts_only_domestic_payment: value.accepts_only_domestic_payment,
		},
	] )
);
```

Reply:
Yep, that's a better approach. Thanks for catching.